### PR TITLE
item best day

### LINKED ISSUE #16: Item best day
GET /api/v1/items/:id/best_day

### PR DESCRIPTION
**Pull request by Dpalazzari**
@tmikeschu 

In this PR:

- Adds best_day test for item
- Adds test at item model level for best_day method
- Adds API endpoint for item best_day
- Adds test for item API endpoint for best_day
- All rspec tests pass
- All relevant spec harness tests pass

Closes #16 

FYI @sallymacnicholas

### CODE REVIEW

File: app/controllers/api/v1/items/best_day_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::Items::BestDayController < ApplicationController
+
+	def show
+		item = Item.find(params[:id])
+		render json: item, serializer: BestDaySerializer
```
Comment (tmikeschu) [inline app/controllers/api/v1/items/best_day_controller.rb:5]: Custom serializer. Nice!

File: app/models/item.rb [hunk 2]
```diff
@@ -5,4 +5,13 @@ class Item < ApplicationRecord
   has_many :invoices, through: :invoice_items
 
   default_scope { order(:id) }
+
+  def best_day
+		invoices
+    .joins(:invoice_items)
+		.group('invoices.id')
+		.order("sum(invoice_items.quantity) DESC, invoices.created_at DESC")
```
Comment (tmikeschu) [inline app/models/item.rb:13]: Great job with the backup ordering parameter.

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -6,4 +6,16 @@
     it { should have_many :invoice_items }
     it { should have_many :invoices }
   end
+
+  context 'business analytics' do
+  	describe '#best_day' do
+  		it 'returns the best day' do
+  			item 		= create(:item)
+  			invoice = create(:invoice)
```
Comment (tmikeschu) [inline spec/models/item_spec.rb:14]: @Dpalazzari can you add another invoice with a different day to verify that it grabs the right day?

**Review by tmikeschu — COMMENTED** [review]
Nice work! See note about spec.

File: spec/models/item_spec.rb (hunk 3 above)
Comment (Dpalazzari) [reply spec/models/item_spec.rb:14]: @tmikeschu Done!

**Pull request merged**